app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -531,41 +531,35 @@ private static void recordSuccessfulSyncTime(String username) {
         //this is the temporary implementation of everything past this point
 
         //Wipe storage
-        //TODO: move table instead. Should be straightforward with sandboxed db's
-        wipeStorageForFourTwelveSync();
+        SQLiteDatabase userDb = CommCareApplication.instance().getUserDbHandle();
+        wipeStorageForFourTwelveSync(userDb);
 
         String failureReason = "";
         try {
-            //Get new data
-            String syncToken = readInput(cache.retrieveCache(), factory);
+            String syncToken = readInputWithoutCommit(cache.retrieveCache(), factory);
             updateUserSyncToken(syncToken);
             Logger.log(LogTypes.TYPE_USER, "Sync Recovery Successful");
+            userDb.setTransactionSuccessful();
             return new Pair<>(PROGRESS_DONE, "");
-        } catch (ActionableInvalidStructureException e) {
-            e.printStackTrace();
-            failureReason = e.getLocalizedMessage();
         } catch (InvalidStructureException | XmlPullParserException
-                | UnfullfilledRequirementsException | SessionUnavailableException | IOException e) {
-            e.printStackTrace();
-            failureReason = e.getMessage();

```

Comment:
You can't delete the setting of this variable because it gets returned at the end of the method

Reply:
Ah, nice catch. Can actually move the return in here though, I think its cleaner